plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/CheckedMetaStoreSupplier.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * *****************************************************************************
+ *
+ *  Pentaho Data Integration
+ *
+ *  Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *  *******************************************************************************
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ *  this file except in compliance with the License. You may obtain a copy of the
+ *  License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * *****************************************************************************
+ *
+ */
+
+package org.pentaho.di.engine.configuration.impl;
+
+import org.pentaho.metastore.api.exceptions.MetaStoreException;
+import org.pentaho.metastore.api.IMetaStore;
+
+
+
+@FunctionalInterface
+public interface CheckedMetaStoreSupplier {
```

Comment:
I can't seem to find the implementation of this interface. Who implements this interface.

Reply:
MetaStoreRunConfigurationFactory and DefaultRunConfigurationProvider take these to pass along a MetaStore. This replaces MetastoreLocator as an API, which didn't allow the checked MetaStoreException to be passed, and really wasn't a great fit. 

It's implemented by a few local lambdas (it is a FunctionalInterface) wherever the above classes are called in this diff.